diff -urp a/opt/atlassian/crowd/apache-tomcat/bin/setenv.sh b/opt/atlassian/crowd/apache-tomcat/bin/setenv.sh
--- a/opt/atlassian/crowd/apache-tomcat/bin/setenv.sh	2018-03-14 09:32:29.786630756 +0800
+++ b/opt/atlassian/crowd/apache-tomcat/bin/setenv.sh	2018-03-10 22:10:03.858412159 +0800
@@ -1,7 +1,11 @@
-JAVA_OPTS="-Xms128m -Xmx512m -Dfile.encoding=UTF-8 $JAVA_OPTS"
+JAVA_OPTS="-Xms${JVM_MINIMUM_MEMORY:-128m} -Xmx${JVM_MAXIMUM_MEMORY:-512m} ${JVM_SUPPORT_RECOMMENDED_ARGS:-"-Datlassian.plugins.enable.wait=300"} -Dfile.encoding=UTF-8 $JAVA_OPTS"
 
 export JAVA_OPTS
 
+CATALINA_OPTS="-Dcrowd.home=${CROWD_HOME:-/var/atlassian/application-data/crowd} -Dcatalina.connector.scheme=${CATALINA_CONNECTOR_SCHEME:-http} -Dcatalina.connector.secure=${CATALINA_CONNECTOR_SECURE:-false} -Dcatalina.connector.proxyName=${CATALINA_CONNECTOR_PROXYNAME:-} -Dcatalina.connector.proxyPort=${CATALINA_CONNECTOR_PROXYPORT:-} -Dcatalina.context.path=${CATALINA_CONTEXT_PATH:-} $CATALINA_OPTS"
+
+export CATALINA_OPTS
+
 # set the location of the pid file
 if [ -z "$CATALINA_PID" ] ; then
     if [ -n "$CATALINA_BASE" ] ; then
diff -urp a/opt/atlassian/crowd/apache-tomcat/conf/server.xml b/opt/atlassian/crowd/apache-tomcat/conf/server.xml
--- a/opt/atlassian/crowd/apache-tomcat/conf/server.xml	2018-03-14 09:32:47.567767816 +0800
+++ b/opt/atlassian/crowd/apache-tomcat/conf/server.xml	2018-03-10 22:10:03.858412159 +0800
@@ -16,10 +16,18 @@
                    URIEncoding="UTF-8"
                    compression="on"
                    sendReasonPhrase="true"
+                   proxyName="${catalina.connector.proxyName}"
+                   proxyPort="${catalina.connector.proxyPort}"
+                   scheme="${catalina.connector.scheme}"
+                   secure="${catalina.connector.secure}"
                    compressableMimeType="text/html,text/xml,application/xml,text/plain,text/css,application/json,application/javascript,application/x-javascript" />
 
         <Engine defaultHost="localhost" name="Catalina">
-            <Host appBase="webapps" autoDeploy="true" name="localhost" unpackWARs="true"/>
+            <Host appBase="webapps" autoDeploy="true" name="localhost" unpackWARs="true">
+                <Context docBase="../../crowd-webapp" path="${catalina.context.path}">
+                    <Manager pathname=""/>
+                </Context>
+            </Host>
         </Engine>
 
         <!-- To connect to an external web server (typically Apache) -->
diff -urp a/opt/atlassian/crowd/start_crowd.sh b/opt/atlassian/crowd/start_crowd.sh
--- a/opt/atlassian/crowd/start_crowd.sh	2018-03-14 09:33:18.179048095 +0800
+++ b/opt/atlassian/crowd/start_crowd.sh	2018-03-10 22:10:03.858412159 +0800
@@ -1,16 +1,28 @@
-#!/bin/sh
+#!/bin/bash
 
+# resolve links - $0 may be a softlink
+PRG="$0"
+while [ -h "$PRG" ]; do
+    ls=`ls -ld "$PRG"`
+    link=`expr "$ls" : '.*-> \(.*\)$'`
+    if expr "$link" : '/.*' > /dev/null; then
+        PRG="$link"
+    else
+        PRG=`dirname "$PRG"`/"$link"
+    fi
+done
+PRGDIR=`dirname "$PRG"`
+
+PRGRUNMODE=false
 if [ "$1" = "-fg" ] || [ "$1" = "run" ]  ; then
-	shift
-	PRGRUNMODE=true
+    shift
+    PRGRUNMODE=true
 else
-	echo "To run Crowd in the foreground, start the server with $0 -fg"
+    echo "To run Crowd in the foreground, start the server with $0 -fg"
 fi
 
-cd "`dirname "$0"`"
-
 if [ "$PRGRUNMODE" == "true" ] ; then
-    exec apache-tomcat/bin/catalina.sh run "$@"
+    exec $PRGDIR/apache-tomcat/bin/catalina.sh run "$@"
 else
-    exec apache-tomcat/bin/startup.sh "$@"
-fi
\ No newline at end of file
+    exec $PRGDIR/apache-tomcat/bin/startup.sh "$@"
+fi
